File: frontend/portfolioERP-web/src/app/features/dashboard/models/dashboard-response.ts
import { OrderStatus } from '../../orders/models/order-status';

export interface OrderStatusSummary {
  status: OrderStatus;
  count: number;
}

export interface RecentOrder {
  id: number;
  orderNumber: string;
  orderDate: string;
  customerName: string;
  status: OrderStatus;
  totalAmount: number;
}

export interface LowStockProduct {
  id: number;
  code: string;
  name: string;
  stockQuantity: number;
}

export interface DashboardResponse {
  activeProducts: number;
  activeCustomers: number;
  totalOrders: number;
  ordersThisMonth: number;
  revenueThisMonth: number;
  monthlyRevenue: MonthlyRevenue[];
  ordersByStatus: OrderStatusSummary[];
  recentOrders: RecentOrder[];
  lowStockProducts: LowStockProduct[];
}
export interface MonthlyRevenue {
  year: number;
  month: number;
  revenue: number;
}

File: backend/PortfolioERP.Application/Features/Dashboard/DashboardResponse.cs
namespace PortfolioERP.Application.Features.Dashboard;

public sealed record DashboardResponse(
    int ActiveProducts,
    int ActiveCustomers,
    int TotalOrders,
    int OrdersThisMonth,
    decimal RevenueThisMonth,
    IReadOnlyList<OrderStatusSummaryResponse> OrdersByStatus,
    IReadOnlyList<RecentOrderResponse> RecentOrders,
    IReadOnlyList<LowStockProductResponse> LowStockProducts,
    IReadOnlyList<MonthlyRevenueResponse> MonthlyRevenue);